controllers/machine_controller.go

```diff
@@ -281,8 +281,16 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 			}
 			r.recorder.Eventf(m, corev1.EventTypeNormal, "SuccessfulDrainNode", "success draining Machine's node %q", m.Status.NodeRef.Name)
 		}
-		logger.Info("Deleting node", "node", m.Status.NodeRef.Name)
 
+		// Verify infrastructure is destroyed before attempting to delete the node.
+		// https://github.com/kubernetes-sigs/cluster-api/issues/2565
+		if ok, err := r.reconcileDeleteExternal(ctx, m); !ok || err != nil {
```

Comment:
This looks like it will not attempt to delete the infrastructure resource if there is not a noderef on the Machine, if it's the last control plane machine, or if there are no control plane machines given the earlier branch for `r.isDeleteNodeAllowed`

Reply:
right thanks! I missed that, let me update...

Reply:
updated and added some units for `isDeleteNodeAllowed`.
What's the reasoning behind proceed deleting the infrastructure when `isDeleteNodeAllowed` is false because of the control plane?